src/test/java/com/internship/socialnetwork/service/PostServiceImplTest.java

```diff
@@ -0,0 +1,190 @@
+package com.internship.socialnetwork.service;
+
+import com.internship.socialnetwork.dto.PostDto;
+import com.internship.socialnetwork.exception.NotFoundException;
+import com.internship.socialnetwork.model.Post;
+import com.internship.socialnetwork.model.User;
+import com.internship.socialnetwork.repository.PostRepository;
+import com.internship.socialnetwork.service.impl.PostServiceImpl;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+public class PostServiceImplTest {
+    private final User author = User.builder()
+            .id(1L)
+            .username("john")
+            .build();
+    private final Post post = Post.builder()
+            .id(1L)
+            .text("Some text")
+            .author(author)
+            .imagePath("somePath")
+            .build();
+
+    private final Post otherPost = Post.builder()
+            .id(2L)
+            .text("other text")
+            .author(author)
+            .imagePath("otherPath")
+            .build();
+
+    private static final String ID_NOT_FOUND = "Post with id: %1$s not found!";
```

Comment:
Pls move this below class declaration

Reply:
Now that you hardcoded 1 you don't need to call String.format when using this constant